src/coreclr/gc/gc.cpp

```diff
@@ -6450,7 +6450,8 @@ class heap_select
         if (GCToOSInterface::CanGetCurrentProcessorNumber())
         {
             uint32_t proc_no = GCToOSInterface::GetCurrentProcessorNumber();
-            proc_no_to_heap_no[proc_no] = (uint16_t)heap_number;
+            // proc_no could likely exceed MAX_SUPPORTED_CPUS on x86
```

Comment:
just making sure I understand when exactly this would happen on x86 when a process is running on a machine that has >64 procs -

+ if the process is running in a cpu group that's not index 0, this would always be a problem because `groupProcNo.GetCombinedValue()` will return a number that's guaranteed to be >64. 

+ if there's no cpu groups (ie, on windows 11), whenever a thread runs on a proc index >= 64, we have this problem.

Reply:
correct, for windows x86 ithe process only gets 32 cores but they could be 0-31 or 64-95, so need to account for that.

Reply:
```suggestion
            // if you are running a 32-bit process on a machine with > 64 procs, even though the process can only use up to 32 procs, the processor index can be >= 64; or in the cpu group case, if the process is not running in cpu group #0, the GetCurrentProcessorNumber will return a number that's >= 64. 
```

Reply:
updated